lib/_http_client.js

```diff
@@ -569,10 +569,20 @@ ClientRequest.prototype.setTimeout = function(msecs, callback) {
 };
 
 ClientRequest.prototype.setNoDelay = function() {
-  this._deferToConnect('setNoDelay', arguments);
+  var argsLength = arguments.length;
+  var args = new Array(argsLength);
+  for (var i = 0; i < argsLength; i++) {
+    args[i] = arguments[i];
+  }
```

Comment:
The size of the statement really just depends on coding style. I could make it the same number of lines as the while loop if I changed it to this:
```js
var argsLength = arguments.length;
var args = new Array(argsLength);
for (var i = 0; i < argsLength; i++) args[i] = arguments[i];
```
Would that style be preferable?